File: T3/codigos2021/T3_Elementos/app/src/main/java/com/borja/t3_elementos/adaptadores/AdaptadorLista.java
package com.borja.t3_elementos.adaptadores;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.borja.t3_elementos.ListasPersoActivity;
import com.borja.t3_elementos.R;
import com.borja.t3_elementos.utils.Coche;

import java.util.ArrayList;

public class AdaptadorLista extends BaseAdapter {

    ArrayList<Coche> listaElementos;
    int[] logos = new int[]{R.drawable.ford,R.drawable.audi};
    Context context;

    public AdaptadorLista(ArrayList lista, Context context){
        this.listaElementos = lista;
        this.context = context;
    }

    // cuantas filas tiene la lista - spinner
    @Override
    public int getCount() {
        return listaElementos.size();
    }

    // retorna el objeto de cada fila
    @Override
    public Object getItem(int position) {
        return listaElementos.get(position);
    }

    // retorna el id del elemento que está en una posición
    @Override
    public long getItemId(int position) {
        return position;
    }

    // retorna la vista de la fila ya pintada
    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        TextView textMarca, textModelo;
        ImageView imagenCoche;

        convertView = LayoutInflater.from(context).inflate(R.layout.item_fila_lista
                ,parent,false);

        Coche cocheActual = listaElementos.get(position);

        textMarca = convertView.findViewById(R.id.marca_lista_perso);
        textModelo = convertView.findViewById(R.id.modelo_lista_perso);
        imagenCoche = convertView.findViewById(R.id.imagen_lista_perso);

        textMarca.setText(cocheActual.getMarca());
        textModelo.setText(cocheActual.getModelo());
        if (cocheActual.getMarca().equals("Mercedes")){
            imagenCoche.setImageResource(R.drawable.sport);
        } else if (cocheActual.getMarca().equals("Audi") || cocheActual.getMarca().equals("Volvo")){
            imagenCoche.setImageResource(R.drawable.luxury);
        } else {
            imagenCoche.setImageResource(R.drawable.normal);
        }

        return convertView;
    }
}

File: Kotlin/T3/codigos2223/T3_Listas_RET/app/src/main/java/com/develop/t3_listas_ret/adapters/AdaptadorLista.kt
package com.develop.t3_listas_ret.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import com.develop.t3_listas_ret.R
import com.develop.t3_listas_ret.model.Usuario

class AdaptadorLista(var lista: ArrayList<Usuario>, var context: Context): BaseAdapter() {

    override fun getCount(): Int {
        return lista.size
    }

    override fun getItem(p0: Int): Usuario {
        return lista.get(p0)
    }

    override fun getItemId(p0: Int): Long {
        return p0.toLong()
    }

    override fun getView(p0: Int, p1: View?, p2: ViewGroup?): View {

        var view: View = LayoutInflater.from(context).inflate(R.layout.list_item,p2,false)

        var imagen: ImageView = view.findViewById(R.id.imagen_fila)
        var nombre: TextView = view.findViewById(R.id.nombre_fila)

        var usuario = lista.get(p0);

        nombre.setText(usuario.nombre)
        if (usuario.genero == "masculino"){
            imagen.setImageResource(R.drawable.male)
        } else if (usuario.genero == "femenino"){
            imagen.setImageResource(R.drawable.female)
        }
        return view;
    }
}